pypesto/visualize/sampling.py

```diff
@@ -91,7 +117,727 @@ def sampling_fval_trace(
+        condition_ids = all_condition_ids
+    if output_ids is None:
+        output_ids = all_output_ids
+
+    # Set default labels for any unspecified labels.
+    labels = {id_: labels.get(id_, id_) for id_ in condition_ids + output_ids}
+
+    if groupby == CONDITION:
+        n_variables = len(output_ids)
+        variable_names = output_ids
+        n_subplots = len(condition_ids)
+    elif groupby == OUTPUT:
+        n_variables = len(condition_ids)
+        variable_names = condition_ids
+        n_subplots = len(output_ids)
+    else:
+        raise ValueError(f'Unsupported groupby value: {groupby}')
+
+    level_opacities, variable_colors = \
+        _handle_colors(levels=levels, n_variables=n_variables)
+
+    if axes is None:
+        n_row = int(np.round(np.sqrt(n_subplots)))
+        n_col = int(np.ceil(n_subplots / n_row))
+        fig, axes = plt.subplots(n_row, n_col, figsize=size, squeeze=False)
+        for ax in axes.flat[n_subplots:]:
+            ax.remove()
+    else:
+        fig = axes.get_figure()
+        if not isinstance(axes, np.ndarray):
+            axes = np.array([[axes]])
+        if len(axes.flat) < n_subplots:
+            raise ValueError(
+                'Provided `axes` contains insufficient subplots. At least '
+                f'{n_subplots} are required.'
+            )
+    artist_padding = (
+        axis_label_padding / (plt.gcf().get_size_inches()*fig.dpi)[0]
+    )
+
+    if groupby == CONDITION:
+        _plot_trajectories_by_condition(
+            summary=summary,
+            condition_ids=condition_ids,
+            output_ids=output_ids,
+            axes=axes,
+            levels=levels,
+            level_opacities=level_opacities,
+            labels=labels,
+            variable_colors=variable_colors,
+        )
+    elif groupby == OUTPUT:
+        _plot_trajectories_by_output(
+            summary=summary,
+            condition_ids=condition_ids,
+            output_ids=output_ids,
+            axes=axes,
+            levels=levels,
+            level_opacities=level_opacities,
+            labels=labels,
+            variable_colors=variable_colors,
+        )
+
+    if title:
+        fig.suptitle(title)
+
+    _handle_legends(
+        fig=fig,
+        axes=axes,
+        levels=levels,
+        labels=labels,
+        level_opacities=level_opacities,
+        variable_names=variable_names,
+        variable_colors=variable_colors,
+        groupby=groupby,
+        artist_padding=artist_padding,
+        n_col=n_col,
+    )
+
+    # X and Y labels
+    xmin = min(ax.get_position().xmin for ax in axes.flat)
+    ymin = min(ax.get_position().ymin for ax in axes.flat)
+    xlabel = (
+        'Cumulative time across all conditions'
+        if groupby == OUTPUT
+        else 'Time'
+    )
+    plt.text(
+        0.5,
+        ymin - artist_padding,
+        xlabel,
+        ha='center',
+        va='center',
+        transform=fig.transFigure
+    )
+    plt.text(
```

Comment:
I think `fig.text()` would make it more easily/intuitively accessible after plotting. Also in line 713.